lib/post.ml

```diff
@@ -22,108 +22,30 @@ type t = {
   feed : Feed.t;
   author : string;
   email : string;
-  content : Nethtml.document list;
+  content : Soup.soup Soup.node;
   mutable link_response : (string, string) result option;
 }
 
-let rec len_prefix_of_html html len =
-  if len <= 0 then (0, [])
-  else
-    match html with
-    | [] -> (len, [])
-    | el :: tl ->
-        let len, prefix_el = len_prefix_of_el el len in
-        let len, prefix_tl = len_prefix_of_html tl len in
-        (len, prefix_el :: prefix_tl)
-
-and len_prefix_of_el el len =
-  match el with
-  | Nethtml.Data d ->
-      let len' = len - String.length d in
-      (len', if len' >= 0 then el else Data (String.sub d 0 len ^ "…"))
-  | Nethtml.Element (tag, args, content) ->
-      (* Remove "id" and "name" to avoid duplicate anchors with the whole
-         post. *)
-      let args = List.filter (fun (n, _) -> n <> "id" && n <> "name") args in
-      let len, prefix_content = len_prefix_of_html content len in
-      (len, Element (tag, args, prefix_content))
-
-let prefix_of_html html len = snd (len_prefix_of_html html len)
-
-let rec filter_map l f =
-  match l with
-  | [] -> []
-  | a :: tl -> (
-      match f a with None -> filter_map tl f | Some a -> a :: filter_map tl f)
-
-let encode_html =
-  Netencoding.Html.encode ~prefer_name:false ~in_enc:`Enc_utf8 ()
-
-let decode_document html = Nethtml.decode ~enc:`Enc_utf8 html
-let encode_document html = Nethtml.encode ~enc:`Enc_utf8 html
```

Comment:
These entity-translating functions are not necessary with Lambda Soup, as Markup.ml applies all the necessary encoding and decoding internally, as required in the HTML5 specification. HTML5 defaults to UTF-8.